components/motor/gpio/motor_encoder.go

```diff
@@ -428,6 +428,21 @@ func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extr
 		return nil
 	}
 
+	if m.loop != nil {
+		positionReached := func(ctx context.Context) (bool, error) {
+			m.stateMu.Lock()
+			goal := m.state.goalPos
+			m.stateMu.Unlock()
+			pos, err := m.position(ctx, extra)
+			return rdkutils.Float64AlmostEqual(pos, goal, 5.0), err
```

Comment:
running into a slight issue here. with high quality encoders, this epsilon can be zero -- the position will get to exactly the goal. but with lower quality encoders like those on the rover, this epsilon needs to be around 5 or `GoFor` will never return. I don't want it to be 5 because that is decreasing the accuracy on nice hardware, but also for low quality hardware it needs to be 5 or it will never return. And I'm also worried with even lower quality hardware it will need to be more than 5 but I don't want to increase the range more.

Reply:
I _think_ it's undershooting but it might be jumping around the goal, so potentially at some point is over. I'll add some logs and see

Reply:
so it does go over, but then usually goes back to a value closer to the goal. I guess either way some accuracy is lost but I think `AlmostEqual` might be better. for example, here's some logs 
```
POS = 1420, GOAL = 1494
POS = 1436, GOAL = 1494
POS = 1432, GOAL = 1494
POS = 1499, GOAL = 1494
POS = 1481, GOAL = 1494
POS = 1499, GOAL = 1494
POS = 1490, GOAL = 1494
...
POS = 1497, GOAL = 1494
POS = 1497, GOAL = 1494
POS = 1492, GOAL = 1494
POS = 1492, GOAL = 1494
POS = 1492, GOAL = 1494
POS = 1492, GOAL = 1494
```
and then it stays at 1492 indefinitely. so it ends up being more accurate than the first time it goes over the goal. what do you think? maybe I can decrease it to 3 instead of 5?